src/openrct2-ui/windows/Ride.cpp

```diff
@@ -6221,13 +6219,14 @@ static void window_ride_income_toggle_secondary_price(rct_window* w)
 
     update_same_price_throughout_flags(shop_item);
 
-    price = ride->price_secondary;
-    game_do_command(0, 1, 0, (1 << 8) | w->number, GAME_COMMAND_SET_RIDE_PRICE, price, 0);
+    auto rideSetPriceAction = RideSetPriceAction(w->number, ride->price_secondary, false);
+    GameActions::Execute(&rideSetPriceAction);
 }
 
 static void window_ride_income_set_primary_price(rct_window* w, money16 price)
```

Comment:
What about `window_ride_income_set_secondary_price`? Have you done a search for all instances of `GAME_COMMAND_SET_RIDE_PRICE`?

Reply:
How did I miss this, i was literally searching the entire solution for the command id.